### DISCUSSION TITLE
workspace/symbol request with empty query does not seem to return all workspace symbols

### DISCUSSION THREAD

**Discussion by catvec** [clangd/clangd#1489] [Q&A]
I am using the clangd language server to help write a tool for a specific code base. I am trying to list all symbols in a workspace. According to the [`workspace/symbol` LSP spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_symbol):

> ```ts
> /**
>  * The parameters of a Workspace Symbol Request.
>  */
> interface WorkspaceSymbolParams extends WorkDoneProgressParams, PartialResultParams {
> 	/**
> 	 * A query string to filter symbols by. Clients may send an empty
> 	 * string here to request all symbols.
> 	 */
> 	query: string;
> }
> ```

Specifically on the `query` field of the request:

>  Clients may send an empty string here to request all symbols.

However sending an empty `query` parameter results in an empty symbols array.

If I send any sort of non-empty `query` with the `workspace/symbol` request (for example `{ "symbol": "A" }`) I get tons of workspace symbols in the response.

I believe I have accounted for all initialization quirks which clangd requires. My workspace has a valid `compile_commands.json` file and I called `textDocument/didOpen` to trigger the background index process, and then waited for the progress notifications for the `backgroundIndexProgress` token to report completion.

Am I correct in assuming that clangd does not provide a bulk symbol list functionality via this `workspace/symbol` request? Or is there some other thing I need to setup to get clangd to respond as I expect?

**Comment by HighCommander4** [comment]
There is some existing discussion about this at https://github.com/clangd/clangd/issues/707, please feel free to chime in there
